### PR TITLE
integration: Test uncompressed DNS messages 

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
This change adds tests to support testing uncompressed DNS messages.

Related: #2591

### CODE REVIEW

**Comment by mqasimsarfraz** [comment]
> I am wondering if trace dns can detect if a DNS message was compressed and if it would make sense to add such a flag to the event.

I am not sure how much work is needed to do that, but I prefer to wait till we get a use-case for it. What do you say?

File: integration/ig/k8s/trace_dns_test.go [hunk 1]
```diff
@@ -24,13 +24,15 @@ import (
 	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
 )
 
-func TestTraceDns(t *testing.T) {
-	t.Parallel()
-	ns := GenerateTestNamespaceName("test-trace-dns")
+func newTraceDnsCmd(t *testing.T, ns string, uncompress bool) *Command {
+	cmdArgs := ""
```
Comment (mauriciovasquezbernal) [inline integration/ig/k8s/trace_dns_test.go:28]: perhaps `newTraceDnsCmd` can accept cmdArgs directly? So we can easily extend this in the future, for instance, to test ipv6.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. Just one minor comment.

We definitely need https://github.com/inspektor-gadget/inspektor-gadget/issues/1271, it's a mess to handle all the duplication of these tests.

Thanks for implementing tests!

File: integration/ig/non-k8s/trace_dns_test.go [hunk 2]
```diff
@@ -152,7 +149,7 @@ func TestTraceDns(t *testing.T) {
 	}
 
 	dnsCmds := []string{
-		"/dnstester & sleep 2", // wait to ensure dns server is running
```
Comment (mqasimsarfraz) [inline integration/ig/non-k8s/trace_dns_test.go:155]: It seems we were suppose to use `&&` here so I fixed it, sharing to make sure it is caught in the review!

Comment (mqasimsarfraz) [reply integration/ig/non-k8s/trace_dns_test.go:155]: Ops, yes you are right. Not sure what was I thinking. :laughing: I will revert the change!

**Comment by mqasimsarfraz** [comment]
> We definitely need https://github.com/inspektor-gadget/inspektor-gadget/issues/1271, it's a mess to handle all the duplication of these tests.

:100:

**Comment by mqasimsarfraz** [comment]
Thanks for the reviews!

**Pull request merged**